apollo-compiler/src/test/graphql/com/example/directive_with_fragment/fragmentsAsDataClasses/TestQuery.kt.expected

```diff
@@ -0,0 +1,99 @@
+// AUTO-GENERATED FILE. DO NOT MODIFY.
+//
+// This class was automatically generated by Apollo GraphQL plugin from the GraphQL queries it found.
+// It should not be modified by hand.
+//
+package com.example.directive_with_fragment
+
+import com.apollographql.apollo3.api.Query
+import com.apollographql.apollo3.api.QueryDocumentMinifier
+import com.apollographql.apollo3.api.ResponseAdapter
+import com.apollographql.apollo3.api.ResponseAdapterCache
+import com.apollographql.apollo3.api.ResponseField
+import com.apollographql.apollo3.api.json.JsonWriter
+import com.example.directive_with_fragment.adapter.TestQuery_Adapter
+import com.example.directive_with_fragment.adapter.TestQuery_ResponseAdapter
+import com.example.directive_with_fragment.fragment.HeroDetails
+import com.example.directive_with_fragment.fragment.HumanDetails
+import kotlin.Boolean
+import kotlin.String
+import kotlin.Suppress
+import kotlin.collections.List
+
+@Suppress("NAME_SHADOWING", "UNUSED_ANONYMOUS_PARAMETER", "LocalVariableName",
+    "RemoveExplicitTypeArguments", "NestedLambdaShadowedImplicitParameter", "PropertyName",
+    "RemoveRedundantQualifierName")
+data class TestQuery(
+  val withDetails: Boolean,
+  val skipHumanDetails: Boolean
+) : Query<TestQuery.Data> {
+  override fun operationId(): String = OPERATION_ID
+
+  override fun queryDocument(): String = QUERY_DOCUMENT
+
+  override fun serializeVariables(writer: JsonWriter, responseAdapterCache: ResponseAdapterCache) {
+    responseAdapterCache.getVariablesAdapterFor(this::class) {
+      TestQuery_Adapter(responseAdapterCache)
+    }.toResponse(writer, this)
+  }
+
+  override fun name(): String = OPERATION_NAME
+
+  override fun adapter(responseAdapterCache: ResponseAdapterCache): ResponseAdapter<Data> {
+    val adapter = responseAdapterCache.getAdapterFor(this::class) {
+      TestQuery_ResponseAdapter(responseAdapterCache)
+    }
+    return adapter
+  }
+
+  override fun responseFields(): List<ResponseField.FieldSet> = listOf(
+    ResponseField.FieldSet(null, TestQuery_ResponseAdapter.RESPONSE_FIELDS)
+  )
+  /**
+   * The query type, represents all of the entry points into our object graph
+   */
+  data class Data(
+    val hero: Hero?
+  ) : Query.Data {
+    /**
+     * A character from the Star Wars universe
+     */
+    data class Hero(
+      val __typename: String = "Character",
+      /**
+       * The ID of the character
+       */
+      val id: String,
+      val asHeroDetails: HeroDetails.Data?,
```

Comment:
I think we used to have `asFoo` for inline fragments and have `fragments.heroDetails` for named fragments.Can we keep that?